sw/device/silicon_creator/lib/ownership/owner_block.c

```diff
@@ -266,40 +314,33 @@ rom_error_t owner_block_flash_apply(const owner_flash_config_t *flash,
           bitfield_field32_read(val, FLASH_CONFIG_LOCK) != kMultiBitBool4False
               ? kHardenedBoolTrue
               : kHardenedBoolFalse;
-      uint32_t region_start = config->start;
-      uint32_t region_end = region_start + config->size;
-
-      // When applying the flash configuration, a region is a ROM_EXT region if
-      // it is exclusively within the ROM_EXT bounds.  This sets the creator
-      // the lockdown policy for a region that is exclusively the creator
-      // region.
-      if (rom_ext_flash_exclusive(region_start, region_end) ==
-          kHardenedBoolTrue) {
-        // Flash region is a ROM_EXT region.  Do nothing.
-        // Some early configurations explicitly set parameters for the ROM_EXT
-        // region.  We ignore these regions in favor of the ROM_EXT setting
-        // its own protection parameters.
-        continue;
-      } else {
-        // Flash region is an owner region.
-        // If the config_side is the same as the owner lockdown side, and
-        // protect_when_primary is requested, deny write/erase to the region.
-        if (config_side == owner_lockdown && pwp != kMultiBitBool4False) {
-          perm.write = kMultiBitBool4False;
-          perm.erase = kMultiBitBool4False;
-        }
-        // If we aren't in a lockdown state, then do not lock the region
-        // configuration via the flash_ctrl regwen bits.
-        if (owner_lockdown == kHardenedBoolFalse) {
-          lock = kHardenedBoolFalse;
-        }
+
+      // Flash region is an owner region.
+      // If the config_side is the same as the owner lockdown side, and
+      // protect_when_primary is requested, deny write/erase to the region.
+      if (config_side == owner_lockdown && pwp != kMultiBitBool4False) {
+        perm.write = kMultiBitBool4False;
+        perm.erase = kMultiBitBool4False;
+      }
+
+      // If we aren't in a lockdown state, then do not lock the region
+      // configuration via the flash_ctrl regwen bits.
+      if (owner_lockdown == kHardenedBoolFalse) {
+        lock = kHardenedBoolFalse;
+      }
+
+      if (*mp_index < 2 * FLASH_CONFIG_REGIONS_PER_SLOT) {
```

Comment:
Is this the MP layout you have in mind? It would be good to capture it in an md doc so that we can provide it as a reference. I am seeing some questions from integration teams.

| MP Slot | Description |
|-|-|
| 0 | `ROM_EXT` slot A |
| 1 | `ROM_EXT` slot B |
| 2 | Owner Slot A - Entry 0 |
| 3 | Owner Slot A - Entry 1 |
| 4 | Owner Slot A - Entry 2 | 
| 5 | Owner Slot B - Entry 0 |
| 6 | Owner Slot B - Entry 1 |
| 7 | Owner Slot B - Entry 2 |

Reply:
Correct: in the `LockedOwner` state, SlotA entries are programmed into MP_REGION registers first, then SlotB entries.

In the unlocked states (`UnlockedAny`, `UnlockedEndorsed`), current_owner's entries are programmed first (for whichever side is primary), then next_owner's entries are programmed (for whichever side is secondary).  This ordering was chosen to keep current_owner's firmware booting in the primary slot and to allow next_owner to program their image into the secondary slot and then attempt a boot with a `SetBl0Next` command.